ospf6d/ospf6_asbr.c

```diff
@@ -2316,6 +2316,31 @@ void ospf6_asbr_redistribute_reset(struct ospf6 *ospf6)
 	}
 }
 
+void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
+{
+	int type;
+	struct ospf6_redist *red;
+
+	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
+		if (type == ZEBRA_ROUTE_OSPF6)
+			continue;
+		red = ospf6_redist_lookup(ospf6, type, 0);
+		if (!red)
+			continue;
+
+		if (type == DEFAULT_ROUTE) {
+			ospf6_redistribute_default_set(
+				ospf6, ospf6->default_originate);
+			continue;
+		}
+
+		ospf6_asbr_redistribute_unset(ospf6, red, type);
+		ospf6_redist_del(ospf6, red, type);
+		red = ospf6_redist_add(ospf6, type, 0);
```

Comment:
Also is there a need to ospf6_redist_del and then ospf6_redist_add, what is the advantage here?

Reply:
fixed
```
		if (ROUTEMAP_NAME(red)) {
			rmap_name = malloc(sizeof(red));
			rmap_name = strdup(ROUTEMAP_NAME(red));
		}
		ospf6_asbr_redistribute_unset(ospf6, red, type);
		if (rmap_name) {
			ospf6_asbr_routemap_set(red, rmap_name);
			free(rmap_name);
		}
		ospf6_asbr_redistribute_set(type, ospf6->vrf_id);
```